common.h

```diff
@@ -35,4 +36,24 @@ static inline uint64_t mmio_read64(void *addr)
 	return ((uint64_t) high << 32) | low;
 }
 
+static inline void mmio_write32(void *addr, uint32_t value)
+{
+	leint32_t *p = addr;
+
+	*p = cpu_to_le32(value);
+}
+
+/* Access 64-bit registers as 2 32-bit if write32 flag set; Some devices fail 64-bit MMIO. */
+static inline void mmio_write64(void *addr, uint64_t value, bool write32)
+{
+	uint64_t *p = addr;
+
+	if (write32) {
+		mmio_write32(addr, value);
+		mmio_write32((uint32_t *)addr + 1, value >> 32);
+		return;
+	}
+
+	*p = cpu_to_le64(value);
+}
```

Comment:
Both function are only used in nvme.c. No need to have them here in the common header.